src/components/PicklistOption/__test__/picklistOption.spec.js

```diff
@@ -0,0 +1,126 @@
+        optionRegisterFn.mockClear();
+    });
+    it('should not register when is variant header', () => {
+        const optionRegisterFn = jest.spyOn(PicklistOption.prototype, 'register');
+        mount(<PicklistOption variant="header" />);
+        expect(optionRegisterFn).not.toHaveBeenCalled();
+        optionRegisterFn.mockClear();
+    });
+    it('should not register when is name passed is not a string', () => {
+        const optionRegisterFn = jest.spyOn(PicklistOption.prototype, 'register');
+        mount(<PicklistOption name={{}} currentValueName="option1" />);
+        expect(optionRegisterFn).not.toHaveBeenCalled();
+        optionRegisterFn.mockClear();
+    });
+    it('should not register when is is selected', () => {
+        const optionRegisterFn = jest.spyOn(PicklistOption.prototype, 'register');
+        mount(<PicklistOption name="option1" currentValueName="option1" />);
+        expect(optionRegisterFn).not.toHaveBeenCalled();
+        optionRegisterFn.mockClear();
+    });
+    it('should register when mount', () => {
+        const optionRegisterFn = jest.spyOn(PicklistOption.prototype, 'register');
+        mount(<PicklistOption name="option1" />);
+        expect(optionRegisterFn).toHaveBeenCalled();
+        optionRegisterFn.mockClear();
+    });
+    it('should unregister when selected', () => {
+        const optionUnregisterFn = jest.spyOn(PicklistOption.prototype, 'unregister');
+        const component = mount(<PicklistOption label="option 1" name="option1" />);
+        component.setProps({ currentValueName: 'option1' });
+        expect(optionUnregisterFn).toHaveBeenCalled();
+        optionUnregisterFn.mockClear();
+    });
+    it('should register when deselected', () => {
+        const component = mount(
+            <PicklistOption label="Option 1" name="option1" currentValueName="option1" />,
+        );
+        const optionRegisterFn = jest.spyOn(PicklistOption.prototype, 'register');
+        component.setProps({ currentValueName: undefined });
+        expect(optionRegisterFn).toHaveBeenCalled();
+        optionRegisterFn.mockClear();
+    });
+    it('should unregister when unmount', () => {
+        const optionUnregisterFn = jest.spyOn(PicklistOption.prototype, 'unregister');
+        const component = mount(<PicklistOption label="option 1" name="option1" />);
+        component.unmount();
+        expect(optionUnregisterFn).toHaveBeenCalled();
+        optionUnregisterFn.mockClear();
+    });
+    it('should fire an event with the right data when click the option', () => {
+        const clickFn = jest.fn();
+        const data = {
+            name: 'option1',
+            label: 'option 1',
+        };
+        const component = shallow(
+            <PicklistOption label="option 1" name="option1" privateOnClick={clickFn} />,
+        );
+        component.find('li').simulate('mousedown');
+        expect(clickFn).toHaveBeenCalledWith(undefined, data);
```

Comment:
test first argument as ```expect.any(object)```
but really we should change this since there is not need to send event